### PR TITLE
#170514634 - Users view bookings

### PR DESCRIPTION
**Pull request by nignanthomas**
#### What does this PR do?
Users view bookings

#### Description of Task to be completed?
- Create actions/reducers.
- Create MyBookings Component.
- Create tests.

#### How should this be manually tested?
- Clone ft-view-bookings-170514634 branch
- run `npm i`
- run `npm run dev`
- Login as a User and click on `My Bookings` in the Menu

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#170514634](https://www.pivotaltracker.com/story/show/170514634)

#### Screenshots (if appropriate)
Screen Shot 2020-01-07 at 14 19 35
Screen Shot 2020-01-07 at 14 17 40
Screen Shot 2020-01-05 at 16 03 10

#### Questions:
N/A

### CODE REVIEW

**Review by Cheza-Dzabala — CHANGES_REQUESTED** [review]
@nignanthomas Really nicely done. One request though, can the bookings be organized into 3. 

1. Pending
2. Approved
3. Rejected

It'll make it easier for the user to navigate.

Question: Are these bookings sorted in any way? 

Secondly, the view single booking page has so much space on it. 
1. Is it necessary seeing as how it does not show any new information? (Opinion Request, not change request)
2. Can it be better placed (Maybe centrally)? (Opinion Request, not change request)

**Review by alainmateso — CHANGES_REQUESTED** [review]
Great job @nignanthomas however, I believe the following changes should be implemented:
- when viewing a specific booking, kindly center the card that has the booking information.
- I believe when viewing a specific booking, it would be better to see the rest of the available information about that booked accommodation (like its image and cost at least). I feel like, it doesn't make any difference viewing a specific booking since all information is already displayed when viewing all bookings.

Kudos for the awesome job

File: src/components/pages/OneBooking.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import Breadcrumbs from '../global/Breadcrumbs';
+
+export class OneBooking extends Component {
+  state = {
+    isLoading: false,
+  };
+
+  componentDidMount = async () => {
+    const { props } = this;
+    const { id } = props.match.params;
```
Comment (nakiwuge) [inline src/components/pages/OneBooking.js:18]: Add a new line

File: src/styles/components/menubar.scss [hunk 2]
```diff
@@ -1,3 +1,9 @@
+// .top-navbar {
```
Comment (nakiwuge) [inline src/styles/components/menubar.scss]: remove commented code

File: src/tests/MyBookingsTest.spec.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    component.setState({ isLoading: false })
+    component.setProps({ status: 'error' });
+    expect(component.exists()).toBe(true);
+  });
+
+  it('Should Mount Bookings Successfully', () => {
+    const component = setUp(mainState);
+    component.setState({ isLoading: false })
+    component.setProps({ status: 'success', bookings: { data: bookingsMocks } });
+    // console.log('component.dive() ===> ', component.find('Component').dive());
```
Comment (nakiwuge) [inline src/tests/MyBookingsTest.spec.js]: remove commented code

File: src/tests/actionsTests/getOneBookingTest.spec.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      type: GET_ONE_BOOKING
+    }];
+    store = mockStore({});
+    await store.dispatch(getOneBooking())
+      .then(async () => {
+        const calledActions = store.getActions();
+        expect(calledActions).toEqual(expectedActions);
+      });
+  });
+});
```
Comment (nakiwuge) [inline src/tests/actionsTests/getOneBookingTest.spec.js]: Add a new line at the end of the file

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@nignanthomas good job on the implementation. kindly work on the feedback below.
- Using tabs for filtering statuses was a good idea but they are not blending in well with the design so kindly user a dropdown to filter through bookings.
- Make the card heading bold and give it the color of the navbar
- There is a mix up in the Menu. The host can now view `My requests` page and they cannot view the `Accommodations` page.
- Link the booking in notifications to the view single booking page.
- Align the card with the All button. The screenshot below shows the cards are not aligned
Screenshot 2020-01-06 at 17 07 58

**Review by Cheza-Dzabala — APPROVED** [review]
LGTM

**Review by alainmateso — APPROVED** [review]

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**